### PR TITLE
feat, refactor: settings window, code cleanup(?)

### LINKED ISSUE #17: Settings Window
The title.

Joking, here is my idea btw:

Proposed Idea

Continued from #16 

Please use this issue for your ideas. I might not be able to add a PR on this, so @Moosems @sumeshir26 I would need your help on this.

### PR DESCRIPTION
**Pull request by im-coder-lg**
The changes I made till now. Need to add user token system, test it out, create a release, and _then_ we remove our token.

Closes #17

### CODE REVIEW

**Comment by im-coder-lg** [comment]
Let's merge this, and I'll try to slowly remove stuff like PYOWM.

**Comment by im-coder-lg** [comment]
Do you guys remember this:

**Comment by im-coder-lg** [comment]
Okay, this PR is ready for now. I'd like to change the config file from `.txt` to `.json`. We could just copy the file content in `utils.py` from TimerX and do this job.
The lines in question are:
https://github.com/Futura-Py/TimerX/blob/5020204afe37872a0f61f540d7b095c8713ea329/utils.py#L9-L42

Since the JSON file is read as a Python dictionary, we could just use key-value pairs. How does that sound?
And I think we must divide the code base. Keep the GUI and API processes in one file, and the rest in another. :shrug: what do you guys think?

**Comment by im-coder-lg** [comment]
Sorry to disturb you guys, but you guys were there on #17; what do you think?

**Review by not-nef — APPROVED** [review]

File: main.py [hunk 1]
```diff
@@ -78,12 +78,13 @@ def __init__(self):
 
             # Apple menus have special names and special commands
             self.app_menu = Menu(self.menubar, tearoff=0, name="apple")
-            self.menubar.add_cascade(label="App", menu=self.app_menu)
+            self.menubar.add_cascade(label="Menu", menu=self.app_menu)
```
Comment (Moosems) [inline main.py:81]: No. This is a mac specific setting  for tk

File: assets/exit.png
Comment (Moosems) [inline assets/exit.png:1]: Please don't use ping icons. Generally not a great idea and it's really not necessary.

File: main.py [hunk 2]
```diff
@@ -245,8 +263,8 @@ def resize_app(self) -> None:
         y_coords = int(self.wm_maxsize()[1] / 2 - minimum_height / 2)
 
         # Place app and make the minimum size the actual minimum size (non-infringable)
-        self.geometry(
-            f"{minimum_width}x{minimum_height}+{x_coords}+{y_coords}")
+        self.geometry(f"{minimum_width}x{
```
Comment (Moosems) [inline main.py:266]: Just put the whole string on its own line instead

File: main.py [hunk 3]
```diff
@@ -309,7 +327,8 @@ def OWMCITY(self, _: Event | None = None) -> None:
 
         # Send request to OpenWeatherMap API
         response: Response = requests_get(
-            f"http://api.openweathermap.org/data/2.5/weather?q={city}&appid={api_key}"
+            f"http://api.openweathermap.org/data/2.5/weather?q={
```
Comment (Moosems) [inline main.py:330]: Same here

File: main.py [hunk 4]
```diff
@@ -335,40 +354,46 @@ def OWMCITY(self, _: Event | None = None) -> None:
             [
                 f"Weather: {weather.status} ~ {weather.detailed_status}",
                 f"Current Temperature: {temperature.get('temp', None):.2f}°C",
-                f"Maximum Temperature: {temperature.get('temp_max', None):.2f}°C",
-                f"Minimum Temperature: {temperature.get('temp_min', None):.2f}°C",
+                f"Maximum Temperature: {
+                    temperature.get('temp_max', None):.2f}°C",
```
Comment (Moosems) [inline main.py:358]: Put the rest of the actual string on its own line. Separate variables and string if you want more than one line

**Review by Moosems — CHANGES_REQUESTED** [review]
A lot of my requests span multiple different points. When going through the changes check all previous changes to see if my comment applies. Otherwise it looks pretty good.

File: main.py (hunk 2 above)
Comment (im-coder-lg) [reply main.py:266]: Well it was due to the Black formatting. I noticed that it does this to long strings.

**Comment by im-coder-lg** [comment]
I'll change the things here. But long strings get separate lines due to the Black formatting.

**Comment by im-coder-lg** [comment]
I think this is fine. Some things, like the button elements still have n early 4-5 lines, but that wouldn't matter, since it's been like that since the start.

**Comment by im-coder-lg** [comment]
AFAIK my changes are done. To me, the PR is ready. But we really must control AutoPEP8 formatting.

In line 349 or something, there is a TODO. I think we finish that when we add more elements, and the window looks like a traditional window, rather than the size we have now. We could try making it into a rather balanced weather app, where it is conventional with some things and unconventional with other things.

**Comment by im-coder-lg** [comment]
Sorry for this sudden surprise, guys, but I can't change more things today. I got some immediate work which I can only complete by tomorrow. I think the PR is ready, ad I approve for merging. If there are other things to change or add, we can do it after merging, if that's fine with all.

**Comment by im-coder-lg** [comment]
@Moosems Idea: rather than a configuration text file, what about a JSON config? Just an idea, very possible, what do you think?

**Comment by Moosems** [comment]
I think it's out of scope for the project :)

Read my comment on the commit I just made by the way.

**Comment by im-coder-lg** [comment]
Hmm... Let's remove autopep8. And maybe we must divide the code into multiple files. And what do you think of a JSON config?

**Comment by Moosems** [comment]
Let's make those separate PR's but I agree with both

**Review by Moosems — APPROVED** [review]
LGTM, let's squash and merge!

**Comment by im-coder-lg** [comment]
lesgoo ppl

**Pull request merged**